EIPS/eip-4910.md

```diff
@@ -0,0 +1,1160 @@
+    uint256 parent;
+    uint256 royaltySplitForItsChildren;
+    string tokentype;
+    uint256 balance;
+    RASubAccount[] rasubaccount;
+}
+```
+##### **[R2]**
+
+*There MUST be a struct for a Royalty Sub-Account linked to a Royalty Account, the minimal set of which is defined below.*
+```
+///
+/// @dev Definition of the struct for a Royalty Sub0Account.
+///
+/// @param isIndividual is the account type identifier, and is a boolean variable, and if set to true, the Royalty Account is that of an individual, if set to false, the Royalty Account is the Royalty Account of the parent NFT
+/// @param royaltysplit gives the royalty percentage for the Royalty Sub-Account as a decimal value smaller than 1 once converted from a uint256 integer
+/// @param royaltybalance is the royalty balance of the Royalty Sub-Account
+/// @param accountid is the address of the owner of the Royalty Sub-Account. If isIndividual is set to true, the address is the address of the owner, and if isIndividual is set to false, the address is defined by the `raAccountId` of the Royalty Account of the parent NFT
+
+struct RASubAccount {
+    bool isIndividual; 
+    uint256 royaltysplit;
+    uint256 royaltybalance;
+    address accountId;
+}
+```
+
+##### **[R3]** 
+
+*The account identifier of a Royalty Account, `raAccountId`, MUST be unique.*
+
+##### **[O1]** 
+
+*The `raAccountId` MAY be defined as the keccak256 hash of `tokenId`, the actual `owner` address, and the current block number, `block.blocknumber`.*
+```
+address memory raAccountId = address(bytes20(keccak256(tokenId,owner,block.blocknumber));
+```
+
+##### **[R4]** 
+
+*The `tokenId` of a NFT MUST be mapped to a `raAccountID` in order to connect an `raAccountId` to a `tokenId`.*
+```
+mapping (uint256 => address) private _tokenindextoRA;
+```
+
+##### **[R5]**
+
+*`raAccountID` MUST be mapped to a `RoyaltyAccount` in order to connect the Royalty  Account identifier to the Royalty Account.*
+```
+mapping (address => RoyaltyAccount) private _royaltyaccount;
+```
+
+#### Rationale: Print (Child) NFTs
+
+The set of requirement to manage Parent-Child NFT Relationships and constraints at each level of the NFT (family) tree e.g. number of children permitted, NFT parents have to be mapped to their immediate NFT children are as follows.
+
+#### Specification: Print (Child) NFTs
+
+##### **[R6]**
+
+*There MUST be a struct that encodes the necessary data for direct parent-child relationships the minimal set of which is defined below.*
+```
+///
+/// @dev Definition of the struct that allows for direct parent-child relationships.
+///
+/// @param maxchildren represents the maximum number of children a parent NFT can have.
+/// @param ancestrylevel represents the level of ancestry of the child NFTs in a generation from the genesis NFT
+/// @param children is the array of children tokenIds of an NFT parent
+struct Child {
+    uint256 maxchildren;
+    uint256 ancestryLevel;
+    uint256[] children;
+}
```

Comment:
Perhaps a dumb question, but I didn't notice `Child` appearing in any public functions/accessors in the spec itself. Does it need to be specified exactly like this, or does an implementation simply have to implement _something_ that enforces the parent-child relationship?

I would assume this also applies to some of the other structs too.

Reply:
@SamWilsn there are no dumb questions, only dumb answers. The naming does not have to be the same as in the spec but the struct elements listed are required for sure. And yes it does apply to other data structures as well -- naming can be changed, struct elements listed are minimally required.

Reply:
As a thought experiment, would an implementation be able to use a `uint8` for `maxChildren`, if that value was never to go above `255`?

I guess what I'm asking is: is this struct (and the similar ones) _necessary_ to specify in the EIP, or should the EIP specify the intended externally visible behaviour and leave the implementation details up to the implementer?

Reply:
@SamWilsn sorry overlooked this comment ... i understand where you are coming from. I think one can argue either way ... if you say, the implementer can choose, then how does a user know, or an application know what the max number supported by an implementation is ... I could easily envision a scenario of only 1 but with 10,000 children ... if this is uint256 and an application is spec-compliant, then there will never be an issue. Sooooo ... this is a fine line, IMHO. I tend to fall on the more prescriptive to reduce ambiguity where it can have significant user impact.